clang/include/clang/Basic/Attr.td

```diff
@@ -1432,6 +1432,22 @@ def SYCLIntelMaxGlobalWorkDim : InheritableAttr {
   let SupportsNonconformingLambdaSyntax = 1;
 }
 
+def SYCLDetailDeviceGlobal: InheritableAttr {
+  let Spellings = [GNU<"device_global">,
+                   CXX11<"__sycl_detail__", "device_global">];
+  let Subjects = SubjectList<[CXXRecord], ErrorDiag>;
+  let LangOpts = [SYCLIsDevice, SilentlyIgnoreSYCLIsHost];
+  let Documentation = [Undocumented];
+}
+
+def SYCLDetailGlobalVariableAllowed : InheritableAttr {
+  let Spellings = [GNU<"global_variable_allowed">,
+                   CXX11<"__sycl_detail__", "global_variable_allowed">];
+  let Subjects = SubjectList<[CXXRecord, GlobalStorageNonLocalVar], ErrorDiag>;
```

Comment:
BTW, I don't think `GlobalStorageNonLocalVar` thing is suitable for us since it checks that subject is a global variable, and the attribute is applied to a record.